src/wrapping.rs

```diff
@@ -0,0 +1,476 @@
+use aes::{
+    cipher::generic_array::GenericArray, Aes128, Aes192, Aes256, BlockDecrypt, BlockEncrypt,
+    NewBlockCipher,
+};
+
+use crate::CryptoBaseError;
+
+const DEFAULT_IV: u64 = 0xa6a6a6a6a6a6a6a6;
+
+/// Build the iv according to the RFC 5649
+///
+/// :data_size: is the size of the key to wrap
+fn build_iv(data_size: usize) -> u64 {
+    let l = 0xA65959A6u32;
+    let r = u32::to_le(data_size as u32);
+
+    (l as u64) << 32 | (r as u64)
+}
+
+/// Check if an iv value is appropriate according to the RFC 5649
+fn check_iv(iv: u64, data: &[u8]) -> bool {
+    let data_size: usize = data.len();
+    if (iv >> 32) as u32 != 0xA65959A6u32 {
+        return false;
+    }
+
+    let real_data_size = (iv as u32) as usize;
+    if real_data_size > data_size || real_data_size as usize <= (data_size - 8) {
+        return false;
+    }
+
+    let paddig_size = data_size - real_data_size;
+    if data[real_data_size..] != vec![0u8; paddig_size] {
+        return false;
+    }
+
+    true
+}
+
+/// Wrap a plain text of variable length
+///
+/// Following RFC 5649
+#[allow(non_camel_case_types)]
+#[allow(non_snake_case)]
```

Comment:
Can we try to get rid of those (for the whole file)?

I understand this is done to stick with RFC, but reading variables with one letter makes the reading really difficult IMO

Reply:
As discuss, we will stick to the RFC variables names